### ISSUE TITLE
CD Onboarding Flow - Direct user to Google Form

### ISSUE THREAD

**Issue by IMHieuVo** [oppia/design-team#113]
**Platform**
Oppia Web

**Describe the request**
When a new contributor visits the contributor dashboard (CD) page at Oppia.org, there is no onboarding provided. 

The aim of this project is to direct logged-in users to a Google Form if they have not yet been accepted as a translator/question contributor yet. This would be implemented as a "Welcome aboard" screen.

**Context for the request**
This feature doesn't really exist currently -- logged-out users just get asked to log in. For logged-in users:
  - They can translate anything immediately (this shouldn't be the case; they should be directed to a signup form and, in general, only given access to languages that they have been granted permission to translate in)
  - They can't see the question submissions tab at all, so they don't know that it exists.

**Link to PRD**
N/A

**Information about deadlines/priorities**
By breaking the project into smaller projects, we can give the developers the design quicker instead of forcing them to wait for the overall project.

**DESIGN APPROVAL CHECKLIST -- DO NOT EDIT.**

- [ ] UX Design approved (Android mocks should include dark mode)
- [ ] UX Writing approved
- [ ] Design Systems Team approved
- [ ] Product approved

_For a design project to be completed, it needs approval from UXD, UXW, DS and PM. Please tag the following reviewers when your project is ready (at least one reviewer for each of the 4 bullet points below):_
- UXD: @rflore / @S4v8n (for Web projects)
- UXW: @stepmalt
- Design systems: @qluuu / @celestinetales
- PM: @seanlip

**Comment by stephanieptdm** [comment]
Previous file https://www.figma.com/file/MH1POsctinme1Xv8WxTcQO/Contributor-Dashboard-Onboarding-Flow-%2342---Intro-Experience?type=design&node-id=1105%3A8299&mode=design&t=78JR39ETV4IDCi3o-1

**Comment by stephanieptdm** [comment]
@tamiorendain Hi Tami! I'd like your feedback on the language I'm using for the Contributors Welcome Page to ensure it aligns with Oppia's UX writing guidelines.
The link for the Figma File it's in the link to mocks.
For now I have the first version, so I would appreciate feedback on it! 
It's on the Section #113 and Contributor Dashboard: Welcome Page Version 01.
Thank you very much!

**Comment by stephanieptdm** [comment]
@tamiorendain  Hi Tami! I'd like your feedback on the language I'm using for the Contributors Welcome Page to ensure it aligns with Oppia's UX writing guidelines.
The link for the Figma File it's in the link to mocks.
For now I have the first version, so I would appreciate feedback on it!
It's on the Section https://github.com/oppia/design-team/issues/113 and Contributor Dashboard: Welcome Page Version 01.
Thank you very much!

**Comment by tamiorendain** [comment]
Hi Stephanie, I've added this to our list :) Thanks!

**Comment by tamiorendain** [comment]
To clarify, would you like us to review only the welcome page, or should we also take a look at the translation page, questions page, and mobile version? @stephanieptdm

**Comment by stephanieptdm** [comment]
@tamiorendain I think maybe you're looking at the previous file.
This is the one:
https://www.figma.com/design/Hl2GYlBn2FjHWrNnrtOldh/Oppia-CD-Onboarding-Flow---Direct-user-to-Google-Form-Mocks-(v01)?node-id=4301-478&t=UPD6ZZmmLOCHTjkP-1 

I updated on the Link to Mocks.

I would like some feedback on the Section named as 113, under the "Contributor Dashboard: Welcome Page Version 01." Title.

**Comment by tamiorendain** [comment]
@stephanieptdm you're right, I was totally in the wrong place haha. Thank you for redirecting!

**Comment by tamiorendain** [comment]
Hi @stephanieptdm, I'm working on this now! Looks quick, so I plan to have UXW comments back to you by tomorrow.

**Comment by tamiorendain** [comment]
Hi again @stephanieptdm, just reviewed the screens and have a few questions!

- What do the translation/question writing forms look like? Do we have screenshots? The current language implies that it's an interest form, but I don't want to assume. Is it a google form?

- What exactly is the flow once users click on a form? I'm assuming that it's:
1. They click on a form
2. A form opens on a new page
3. User must click the "next" arrow to get to the "onboarding 3" screen?

- Will contributors only see this screen once? What if they want to change their choice or add to it?

I've started making comments, but please disregard as I'd like to get this info before I finalize any edits. Thank you!

**Comment by stephanieptdm** [comment]
Hi @Tami Orendain ***@***.***> ,

I actually haven't seen the form yet. @kurin tu ***@***.***> Can
you send it to us? But as far as I know it is a Google Form.

Yes, this is the correct flow of the Version 2, exactly. I'll try to make
it clear.

As for them seeing it only once and in case they want to change their
choice or add to it, I don't have that information. Maybe Kurin could also
help with that.

If you need anything else, please let me know.

Best,
Stephanie

**Comment by seanlip** [comment]
Here is the translation volunteer interest form: https://forms.gle/tssUWJYNsaqub2PF9

Here is the practice questions volunteer interest form: https://forms.gle/E2V72ZPh1pGUS4SG9

I can't speak to the flow in the mocks (will defer to @stephanieptdm for that), but as for the form: after prospective volunteers click on a form, they fill it in and submit it. They then wait to get a response from the translations/questions coordinators. The coordinators will email the volunteer with the results of the decision (accept/reject). If accept, they will add them to the corresponding translation/questions team and send them a Google Doc guide for volunteers.

Note that the form lives outside the Oppia application and the two don't communicate with each other.

I hope this helps, lmk if there's anything I can clarify (and also @stephanieptdm please feel free to ask if there are key parts of the process you need more info on, too). Thanks!

**Comment by tamiorendain** [comment]
Ahh got it, thank you @seanlip. Would it be right to say, then, that these screens aren't a "Contributor Dashboard," since they technically aren't contributors yet? When I hear "dashboard," I think of the hub/homepage where contributors go to start their work. This seems more like a "contributor interest page". Does that sound right? Can we change the name of the page?

**Comment by seanlip** [comment]
Hm, I think the issue is that it has the same URL as the contributor dashboard. The idea is that when a user who doesn't have the necessary permissions goes there, they see this screen or interstitial; a user who already does have the permissions doesn't see it.

So, the page is still the contributor dashboard page, and the user is someone who sees this on the Oppia site (e.g. mentioned in the navbar as a way for folks to contribute), and visits it. But, given the above, changing the name would cause some confusion, I think?

**Comment by tamiorendain** [comment]
Ah, interesting - so it's a dashboard, but access is limited until they fill out the form and get accepted.

Follow up question @seanlip : You mentioned that the user can get to this page through the Oppia site, but which link would they click to get there? All volunteer links I can see currently lead to this non-developer volunteer form: https://docs.google.com/forms/d/e/1FAIpQLSc5_rwUjugT_Jt_EB49_zAKWVY68I3fTXF5w9b5faIk7rL6yg/viewform

**Comment by seanlip** [comment]
Ah, we plan to add a link to it in the user navbar dropdown (at least) once we have this onboarding flow in place, to give it more visibility. Currently we don't quite want to do that to avoid a spammy flood of translations (since atm there is no gating).

**Comment by tamiorendain** [comment]
Got it. Thank you for the extra info, @seanlip. Would it ever be possible for the dropdown links to lead directly to the forms? I feel like that would be a quicker, cleaner user experience. I'll keep editing with all the original info in mind, though :)

@stephanieptdm, I've made some copy edits based on this conversation! @seanlip I also tagged you in a question in the Figma, but I'll also leave these questions here:

- I wasn't sure whether or not we expect that most applicants will be accepted. Did we want the language to say, "submitting a form means you're a volunteer automatically," or, "submit a form and we'll get back to you"? I left a couple of options for both the opening and ending text - would love to discuss which perspective is more appropriate.

- Can we change the titles of the google forms? I'd like to update them to be consistent with the dashboard, so the new titles would be: "Oppia volunteer translator interest form" and "Oppia volunteer question writer interest form"

**Comment by seanlip** [comment]
Hi @tamiorendain, I think it is totally fine for the dropdown links to lead directly to the forms, though I am not sure if it would surprise the user to be taken to a non-Oppia page when they're selecting something from the dropdown menu. Happy to defer to you and @stephanieptdm to decide on that.

Re the questions:

- It's the latter. Their application needs to be approved first, and we may reject it if their submitted example is not strong enough.
- Yup, I've already done this. Thanks!

**Comment by tamiorendain** [comment]
I think it wouldn't be too surprising, depending on how the screen and UX are mocked up! Happy to have a longer discussion about what that could look like. For now, though, I think the UXW review is finishing up on this project. Thank you for updating the forms as well!

@stephanieptdm - I've left comments with the final copy suggestions in the Figma. Let me know if there's anything you want me to explain!

**Comment by stephanieptdm** [comment]
Hi, @Sean Lip ***@***.***> and @Tami Orendain ***@***.***>

Thank you for clarifying the project details and for your feedback on
Figma—it was very helpful!
I'll review everything and get back to you once I’ve completed all the
modifications. If there’s anything I’m unsure about, I’ll be sure to reach
out.

Best,
Stephanie

**Comment by stephanieptdm** [comment]
Hi @Sean Lip ***@***.***>,

I was just checking the comments on Figma, and Russell raised a good
question: Do we have an estimated time to follow up with the user about the
forms?

**Comment by seanlip** [comment]
That's a good question. It should typically be within 2 days, but for now, let's say 1 week.

Thanks!

**Comment by stephanieptdm** [comment]
Ok, so we may set a one-week deadline to align with their expectations. Thanks,
Sean!

**Comment by stephanieptdm** [comment]
Hi everyone,

I’ve updated the Figma file and created a new version based on your
feedback.
Please take a look and let me know if everything looks good!

Best,

Stephanie

**Comment by stephanieptdm** [comment]
@davishedrick Hi Davisi! I'd like your feedback on the language I'm using for the Contributors Welcome Page to ensure it aligns with Oppia's UX writing guidelines.
Tami had already reviewed and approved the previous version, but Sean made a UXW-related observation on the latest mockup.
Here’s the link that takes you directly to Sean’s comment on the latest mockup: https://www.figma.com/design/COmfMAlx9ixlqhF5DxT0yO?node-id=168-3400#1225683767

Thank you very much!
